core/src/com/unciv/logic/civilization/TechManager.kt

```diff
@@ -109,8 +108,23 @@ class TechManager {
         return if (techsToResearch.isEmpty()) null else techsToResearch[0]
     }
 
-    private fun researchOfTech(TechName: String?): Int {
-        return if (techsInProgress.containsKey(TechName)) techsInProgress[TechName]!! else 0
+    fun researchOfTech(TechName: String?) = techsInProgress[TechName] ?: 0
+    // Was once duplicated as fun scienceSpentOnTech(tech: String): Int
+    
+    /** Replace a changed tech name, only temporarily used for breaking ruleset updates */
+    fun replaceUpdatedTechName(oldTechName: String, newTechName: String) {
+        if (oldTechName in techsResearched) {
+            techsResearched.remove(oldTechName)
+            techsResearched.add(newTechName)
+        }
+        val index = techsToResearch.indexOf(oldTechName)
+        if (index >= 0) {
+            techsToResearch[index] = newTechName
+        }
+        if (oldTechName in techsInProgress) {
+            techsInProgress[newTechName] = researchOfTech(oldTechName)
+            techsInProgress.remove(oldTechName)
+        }
     }
```

Comment:
Please put ALL state-changing-as-a-result-of-update code in GameInfo.kt (or alternatively we could create a new file that will deal just with that)

Reply:
> Dedicated update compatibility module

***Excellent*** idea!

`<mumbling>`... and move all existing ... would be a branch of its own ...`</mumbling>`

Reply:
You're right, that's for a different PR :)
For now just move this to GameInfo